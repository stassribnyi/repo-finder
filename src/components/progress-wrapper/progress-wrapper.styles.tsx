import styled, { css } from 'styled-components';
import { LinearProgress } from '@material-ui/core';

const Progress = styled(LinearProgress)`
  position: absolute;

  ${({ theme }) => css`
    top: ${theme.spacing(1)}px;
    left: ${theme.spacing(1)}px;
    width: calc(100% - ${theme.spacing(2)}px);
  `}
`;

const Container = styled.div`
  width: 100%;
  position: relative;

  padding: ${({ theme }) => theme.spacing(1)}px 0;
`;

export const Styled = { Container, Progress };
