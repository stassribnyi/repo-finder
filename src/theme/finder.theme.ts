import { unstable_createMuiStrictModeTheme as createMuiTheme } from '@material-ui/core';

export const FINDER_THEME = createMuiTheme({
  breakpoints: {
    values: {
      tablet: 640,
      laptop: 1024,
      desktop: 1280,
    },
  } as any, // FIXME: Use module augmentation
  overrides: {
    MuiCssBaseline: {
      '@global': {
        html: {
          WebkitFontSmoothing: 'auto',
        },
      },
    },
  },
});
