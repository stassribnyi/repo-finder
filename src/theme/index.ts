import { unstable_createMuiStrictModeTheme as createMuiTheme } from '@material-ui/core';

export const FINDER_THEME = createMuiTheme({
  overrides: {
    MuiCssBaseline: {
      '@global': {
        html: {
          WebkitFontSmoothing: 'auto',
        },
      },
    },
  },
});
