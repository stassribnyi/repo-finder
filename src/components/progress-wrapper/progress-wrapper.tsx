import React from 'react';
import { Styled } from './progress-wrapper.styles';

export type ProgressWrapperProps = Readonly<{
  isLoading: boolean;
  showContent?: boolean;
}>;

export const ProgressWrapper: React.FC<ProgressWrapperProps> = ({
  isLoading,
  children,
  showContent = false,
}) => (
  <Styled.Container>
    {isLoading ? (
      <>
        <Styled.Progress color='secondary' />
        {showContent && children}
      </>
    ) : (
      children
    )}
  </Styled.Container>
);
