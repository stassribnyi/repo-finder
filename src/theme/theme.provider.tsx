import React from 'react';

import {
  CssBaseline,
  ThemeProvider as MuiThemeProvider,
} from '@material-ui/core';
import { ThemeProvider as SCThemeProvider } from 'styled-components';

import { FINDER_THEME } from './finder.theme';

export const FinderThemeProvider: React.FC = ({ children }) => (
  <MuiThemeProvider theme={FINDER_THEME}>
    <SCThemeProvider theme={FINDER_THEME}>
      <CssBaseline />
      {children}
    </SCThemeProvider>
  </MuiThemeProvider>
);
