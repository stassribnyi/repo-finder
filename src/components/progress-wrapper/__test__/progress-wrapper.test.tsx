import React from 'react';
import { cleanup, render } from '@testing-library/react';

import { ProgressWrapper } from '../progress-wrapper';
import { FinderThemeProvider } from '../../../theme';

afterEach(cleanup);

describe('ProgressWrapper', () => {
  it('should show progress bar', async () => {
    // Arrange
    const { asFragment, queryByText } = render(
      <FinderThemeProvider>
        <ProgressWrapper isLoading={true}>
          <p>progress without content</p>
        </ProgressWrapper>
      </FinderThemeProvider>
    );

    // Act
    const contentElement = queryByText(/progress without content/i);

    // Assert
    expect(contentElement).toBeNull();
    expect(asFragment()).toMatchSnapshot('progress without content');
  });

  it('should show progress bar alongside with content', async () => {
    // Arrange
    const { asFragment, queryByText } = render(
      <FinderThemeProvider>
        <ProgressWrapper isLoading={true} showContent={true}>
          <p>progress with content</p>
        </ProgressWrapper>
      </FinderThemeProvider>
    );

    // Act
    const contentElement = queryByText(/progress with content/i);

    // Assert
    expect(contentElement).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot('progress with content');
  });

  it('should show progress bar alongside with content', async () => {
    // Arrange
    const { asFragment, queryByText } = render(
      <FinderThemeProvider>
        <ProgressWrapper isLoading={false}>
          <p>content only</p>
        </ProgressWrapper>
      </FinderThemeProvider>
    );

    // Act
    const contentElement = queryByText(/content only/i);

    // Assert
    expect(contentElement).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot('content only');
  });
});
