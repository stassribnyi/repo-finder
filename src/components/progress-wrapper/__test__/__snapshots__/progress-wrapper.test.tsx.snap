// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressWrapper should show progress bar alongside with content: content only 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  position: relative;
  padding: 8px 0;
}

<div
    class="c0"
  >
    <p>
      content only
    </p>
  </div>
</DocumentFragment>
`;

exports[`ProgressWrapper should show progress bar alongside with content: progress with content 1`] = `
<DocumentFragment>
  .c1 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: calc(100% - 16px);
}

.c0 {
  width: 100%;
  position: relative;
  padding: 8px 0;
}

<div
    class="c0"
  >
    <div
      class="MuiLinearProgress-root MuiLinearProgress-colorSecondary c1 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        class="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Indeterminate"
      />
      <div
        class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorSecondary"
      />
    </div>
    <p>
      progress with content
    </p>
  </div>
</DocumentFragment>
`;

exports[`ProgressWrapper should show progress bar: progress without content 1`] = `
<DocumentFragment>
  .c1 {
  position: absolute;
  top: 8px;
  left: 8px;
  width: calc(100% - 16px);
}

.c0 {
  width: 100%;
  position: relative;
  padding: 8px 0;
}

<div
    class="c0"
  >
    <div
      class="MuiLinearProgress-root MuiLinearProgress-colorSecondary c1 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        class="MuiLinearProgress-bar MuiLinearProgress-barColorSecondary MuiLinearProgress-bar1Indeterminate"
      />
      <div
        class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorSecondary"
      />
    </div>
  </div>
</DocumentFragment>
`;
